### PR TITLE
Explicitly add the current principal to the acl in case of group sharing

### LINKED ISSUE #23011: Group calendars can't be edited by group members
A calendar that is shared with a group (group is allowed to edit) cannot be edited by the group members. They can only see the already entered dates. To make a calendar editable by group members, it has to be shared with each one of them individually.

 
### Steps to reproduce
1. Create a group with some users.
2. Create a calendar.
3. Share this calendar with the group and make it group editable.

### Expected behaviour
The group members should see the shared calendar and be able to insert new dates or change already entered ones.

### Actual behaviour
The calendar is shared, but read-only.

### Server configuration
**Operating system**: Debian 3.16.0-4-amd64

**Web server:** Apache 2.4.10

**Database:** MySQL 5.5.47

**PHP version:** 5.6.27

**ownCloud version:** 9.0.0

**Updated from an older ownCloud or fresh install:** update from 8.2

**Where did you install ownCloud from:**

**Signing status (ownCloud 9.0 and above):**

**List of activated apps:**
Enabled:
  - activity: 2.2.1
  - calendar: 1.0
  - comments: 0.2
  - dav: 0.1.5
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - templateeditor: 0.1
  - updatenotification: 0.1.0

Disabled:
  - encryption
  - external
  - files_external
  - user_external
  - user_ldap

**Are you using external storage, if yes which one:** No

**Are you using encryption:** No

**Are you using an external user-backend, if yes which one:** No

### PR DESCRIPTION
**Pull request by DeepDiver1975**
fixes #23011
refs https://github.com/owncloud/core/issues/22678#issuecomment-195490303

@karlitschek requesting backport - otherwise group shared calendars and addressbooks are not editable
# Test Scenario
1. share a calendar with a group with write permissions
2. login with a user which is member of that group
3. that user should be capable to edit events

BEFORE:
bildschirmfoto von 2016-03-24 10-19-06

AFTER:
bildschirmfoto von 2016-03-24 10-17-46

### CODE REVIEW

**Comment by karlitschek** [comment]
makes sense. Please backport!

File: apps/dav/lib/caldav/caldavbackend.php [hunk 1]
```diff
@@ -106,10 +107,10 @@ class CalDavBackend extends AbstractBackend implements SyncSupport, Subscription
 	 * @param IDBConnection $db
 	 * @param Principal $principalBackend
 	 */
-	public function __construct(IDBConnection $db, Principal $principalBackend) {
+	public function __construct(IDBConnection $db, $userPrincipalBackend, GroupPrincipalBackend $groupPrincipalBackend) {
```
Comment (nickvergessen) [inline apps/dav/lib/caldav/caldavbackend.php]: missing in docs

File: apps/dav/lib/carddav/carddavbackend.php [hunk 2]
```diff
@@ -77,12 +78,13 @@ class CardDavBackend implements BackendInterface, SyncSupport {
 	 * @param EventDispatcherInterface $dispatcher
 	 */
 	public function __construct(IDBConnection $db,
-								Principal $principalBackend,
+								Principal $userPrincipalBackend,
+								GroupPrincipalBackend $groupPrincipalBackend,
```
Comment (nickvergessen) [inline apps/dav/lib/carddav/carddavbackend.php]: missing in docs

File: apps/dav/lib/dav/sharing/backend.php [hunk 3]
```diff
@@ -43,9 +46,10 @@ class Backend {
 	 * @param Principal $principalBackend
 	 * @param string $resourceType
 	 */
-	public function __construct(IDBConnection $db, Principal $principalBackend, $resourceType) {
+	public function __construct(IDBConnection $db, Principal $userPrincipalBackend, GroupPrincipalBackend $groupPrincipalBackend, $resourceType) {
```
Comment (nickvergessen) [inline apps/dav/lib/dav/sharing/backend.php]: missing in docs

**Comment by PVince81** [comment]
@DeepDiver1975 still "Developing" ?

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 still "Developing" ?

yes

File: apps/dav/lib/carddav/carddavbackend.php (hunk 2 above)
Comment (DeepDiver1975) [inline apps/dav/lib/carddav/carddavbackend.php]: addressed

**Comment by PVince81** [comment]
Code looks good :+1:

**Comment by DeepDiver1975** [comment]
rebase fuckup :angry:

**Comment by MorrisJobke** [comment]
Already works for me in master.

**Comment by MorrisJobke** [comment]
This is the fixing commit: 06e8c70400dd6e3e1c153c4258eba0357c502a7f

**Comment by MorrisJobke** [comment]
> This is the fixing commit: 06e8c70

https://github.com/owncloud/core/pull/23342

**Comment by DeepDiver1975** [comment]
ok - let's close this one - THX

**Pull request closed without merging**